src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -1697,6 +1696,15 @@ private bool SkipSingleLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
                 return false;
             }
 
+            // If we have encountered \r, check and advance index if next character is \n
+            if (localBuffer[idx] == JsonConstants.CarriageReturn 
+                && idx < localBuffer.Length - 1
```

Comment:
If `idx + 1 >= localBuffer.Length`, we don't know if the next byte is a `\n` and whether we should include it in the comment or not.

Should we return false in this case (unless `IsLastSpan` is `true`)?